R/jira.R

```diff
@@ -1077,52 +1056,150 @@ download_jira_issues_comments_by_issuekey <- function(domain,
                                                   credentials,
                                                   jql_query,
                                                   fields,
                                                   save_path_issue_tracker_issues,
-                                                  maxresults,
+                                                  maxResults,
                                                   verbose,
-                                                  maxdownloads,
-                                                  file_name,
+                                                  maxDownloads,
                                                   unaltered_file_path){
 
-  issue_refresh <- paste0(unaltered_file_path, file_name)
+  # Check if the file exists
+  if(file.exists(unaltered_file_path)) {
+    # Check if the file is empty by checking its size
+    # List all files and subdirectories in the directory
+    contents <- list.files(path = unaltered_file_path)
+
+    # If the file is empty, download all issues
+    if(length(contents) == 0) {
+      message("The file is empty.\n")
+      download_and_save_jira_issues(domain,
+                                    credentials,
+                                    jql_query,
+                                    fields,
+                                    save_path_issue_tracker_issues,
+                                    maxResults,
+                                    verbose,
+                                    maxDownloads,
+                                    search_query = NULL
+                                    )
+    } else {
+      message("File size: ", file.size(unaltered_file_path))
+      # If file is not empty, run the refresh function
+      # Get the file name with the latest 'issueKey' value
+      file_name_with_greatest_issue_key <- parse_jira_latest_date(unaltered_file_path)
+      message("Filename with latest date: ", file_name_with_greatest_issue_key)
+
+      # Prep the path and filename
+      issue_refresh <- paste0(unaltered_file_path, file_name_with_greatest_issue_key)
+
+      # Read the JSON file
+      json_data <- jsonlite::fromJSON(txt = issue_refresh, simplifyVector = FALSE)
+
+      # Extract the Maximum issue key value
+      # Start with a low value assuming no negative numbers
+      max_numeric_part <- -1
+      max_key <- ""
+
+      for (issue in json_data$issues) {
+        # Extract the key for the current issue
+        current_key <- issue$key
+
+        # Extract the numeric part of the key
+        # Assuming the key format is "PROJECTNAME-NUMBER"
+        numeric_part <- as.numeric(sub("^[^-]+-", "", current_key))
+
+        # Check if the numeric part is greater than the current maximum
+        if (numeric_part > max_numeric_part) {
+          # Update the maximum numeric part and the corresponding key
+          max_numeric_part <- numeric_part
+          max_key <- current_key
+        }
+      }
```

Comment:
What should be moved inside your refresh function is the function call to @ian-lastname function. You moved its content instead. Can you replace by the function call? The logic will be confusing to follow otherwise.